### PR TITLE
Skip recommendation

### PR DESCRIPTION
**Pull request by supersweet-dev**
Adds a "next recommendation" button under Recommended Action. Shows next possible suggestion.

### CODE REVIEW

File: client/src/components/my-mentor-card.tsx [hunk 1]
```diff
@@ -261,16 +253,14 @@ export default function MyMentorCard(props: {
...
                 <Typography
                   variant="h6"
                   color="textSecondary"
                   align="left"
                   data-cy="mentor-card-type"
                 >
-                  {props.type[0].toUpperCase() +
-                    props.type.slice(1).toLowerCase()}{" "}
-                  Mentor
+                  {type[0].toUpperCase() + type.slice(1).toLowerCase()} Mentor
```
Comment (beatthat) [inline client/src/components/my-mentor-card.tsx:263]: let's have a util function like `toTitleCase` and then import and use it here

File: client/src/components/recommended-action-button.tsx [hunk 2]
```diff
@@ -19,28 +19,32 @@ function urlBuild(base: string, params: Record<string, string>) {
...
-  });
-  const idle = mentor?.answers.find(
+  const idle = props.mentor?.answers.find(
     (a) => a.question.name === UtteranceName.IDLE
   );
-  const categories: {
+  const idleIncomplete = idle?.status === Status.INCOMPLETE;
+  const isVideo = props.mentor?.mentorType === MentorType.VIDEO;
+  const thumbnail = props.mentor?.thumbnail;
+  interface category {
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:30]: interfaces/types always Pascal case (Category)

File: client/src/components/recommended-action-button.tsx [hunk 3]
```diff
@@ -62,102 +67,204 @@ export default function RecommendedActionButton(props: {
       });
     });
   });
+  const incompleteRequirement = categories
+    .filter((c) => c.isRequired)
+    .find((c) => c.answers.find((a) => a.status === Status.INCOMPLETE));
+  console.log;
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:73]: delete `console.log`

File: client/src/components/recommended-action-button.tsx [hunk 4]
```diff
@@ -62,102 +67,204 @@ export default function RecommendedActionButton(props: {
...
     reason: string;
     action: () => void;
-  } {
-    if (mentor?.thumbnail == "")
+    skippable: boolean;
+    skip: Conditions;
+  }
+
+  function recommend(conditions: Conditions): Recommendation {
+    if (conditions.thumbnail == "")
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:103]: ```if (!thumbnail)```

File: client/src/components/recommended-action-button.tsx [hunk 5]
```diff
@@ -62,102 +67,204 @@ export default function RecommendedActionButton(props: {
...
       return {
         text: "Add a Thumbnail",
         reason: "A thumbnail helps a user identify your mentor",
         action: () => undefined,
+        skippable: true,
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:108]: not new, but the return type of `action` above should be `void`

File: client/src/components/recommended-action-button.tsx [hunk 6]
```diff
@@ -62,102 +67,204 @@ export default function RecommendedActionButton(props: {
...
-        data-cy="recommended-action-upload"
-        type="file"
-        onChange={(e) => {
-          e.target.files instanceof FileList
-            ? props.setThumbnail(e.target.files[0])
-            : undefined;
-        }}
-      />
-      <label htmlFor="thumbnail-upload">
+      {props.mentor?.thumbnail == "" ? (
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:208]: ```Boolean(props.mentor?.thumbnail)```

File: cypress/cypress/integration/mentor-card.spec.ts [hunk 7]
```diff
@@ -153,22 +153,108 @@ describe("My Mentor Page", () => {
       cy.url().should("include", "videoId=idletest");
     });
 
-    it("Asks user with thumbnail and idle video to answer incomplete questions", () => {
+    it("Asks user with incomplete Required Subjects to finish them.", () => {
```
Comment (beatthat) [inline cypress/cypress/integration/mentor-card.spec.ts:156]: why capitalize `Required Subjects` here?

**Review by beatthat — CHANGES_REQUESTED** [review]
looks good. Very minor changes requested inline

File: client/src/components/recommended-action-button.tsx (hunk 5 above)
Comment (supersweet-dev) [reply client/src/components/recommended-action-button.tsx:109]: Last push addresses most changes except this one. What's the proper way to do this?

File: client/src/components/recommended-action-button.tsx [hunk 8]
```diff
@@ -19,28 +19,32 @@ function urlBuild(base: string, params: Record<string, string>) {
...
-    subject: "",
-  });
-  const idle = mentor?.answers.find(
+  const idle = props.mentor?.answers.find(
     (a) => a.question.name === UtteranceName.IDLE
   );
-  const categories: {
+  const idleIncomplete = idle?.status === Status.INCOMPLETE;
+  const isVideo = props.mentor?.mentorType === MentorType.VIDEO;
+  const thumbnail = props.mentor?.thumbnail != "";
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:29]: is this really `hasThumbnail`? Fix name and use truthiness/falsiness 

```typescript
const hasThumbnail = Boolean(props.mentor?.thumbnail)
```

File: client/src/components/recommended-action-button.tsx [hunk 9]
```diff
@@ -7,8 +7,8 @@ The full terms of this copyright and license should always be found in the root
 import { Button, Typography } from "@material-ui/core";
 import React from "react";
 import { navigate } from "gatsby";
-import { useWithReviewAnswerState } from "hooks/graphql/use-with-review-answer-state";
-import { Answer, MentorType, Status, UtteranceName } from "types";
+import { Answer, Mentor, MentorType, Status, UtteranceName } from "types";
```
Comment (beatthat) [inline client/src/components/recommended-action-button.tsx:10]: We need to get out of habit of throwing tons of model logic into our components (presentation)

Can this be a folder `./components/recommended-action/`

...and then two files within:

 - recommended-action-button.tsx (just the presentation)
 - use-recommended-action.ts (should not need the tsx extension, all the data and logic are here)

File: client/src/components/recommended-action-button.tsx (hunk 5 above)
Comment (beatthat) [reply client/src/components/recommended-action-button.tsx:109]: I think can just be `action: () => void` does that not work?

File: client/src/helpers.ts [hunk 10]
```diff
@@ -23,3 +23,6 @@ export function copyAndMove<T>(a: T[], moveFrom: number, moveTo: number): T[] {
   const removed = copyAndRemove(a, moveFrom);
   return [...removed.slice(0, moveTo), item, ...removed.slice(moveTo)];
 }
+export function toTitleCase(convert: string) {
```
Comment (beatthat) [inline client/src/helpers.ts:26]: ```export function toTitleCase(convert: string): string```

**Review by beatthat — CHANGES_REQUESTED** [review]
looks good and close. Main thing I'd like to see is whether we can break up `recommended-action` into a component (just presentation) and a hook (all the data and logic and no jsx)

File: client/src/components/recommended-action-button.tsx (hunk 9 above)
Comment (supersweet-dev) [reply client/src/components/recommended-action-button.tsx:10]: ended up putting this and mentor card in a folder, split off sub-components and logic functions

File: client/src/components/recommended-action-button.tsx (hunk 5 above)
Comment (supersweet-dev) [reply client/src/components/recommended-action-button.tsx]: nope, doesn't seem to like that

Comment (beatthat) [reply client/src/components/recommended-action-button.tsx]: oh sorry that's the actual function; i thought it was a type spec. I think this is more standard:  `action: () => {}`

**Review by beatthat — DISMISSED** [review]
looks great

**Comment by supersweet-dev** [comment]
pushed merge with main

**Review by beatthat — APPROVED** [review]

**Pull request merged**